src/Compilers/Core/Portable/FileSystem/PathUtilities.cs

```diff
@@ -436,6 +436,33 @@ public static string CombinePathsUnchecked(string root, string? relativePath)
             return root + relativePath;
         }
 
+        /// <summary>
+        /// Combines paths with the same semantics as <see cref="Path.Combine(string, string)"/>
+        /// but does not throw on null paths or paths with invalid characters.
+        /// </summary>
+        /// <param name="root">First path: absolute, relative, or null.</param>
+        /// <param name="path">Second path: absolute, relative, or null.</param>
+        /// <returns>
+        /// The combined paths. If <paramref name="path"/> contains an absolute path, returns <paramref name="root"/>.
+        /// </returns>
+        /// <remarks>
+        /// Relative and absolute paths treated the same as <see cref="Path.Combine(string, string)"/>.
+        /// </remarks>
+        public static string? CombinePaths(string? root, string? path)
```

Comment:
```suggestion
       [NotNullIfNotNull("path")]
        public static string? CombinePaths(string? root, string? path)
```

Believe this accurately captures the semantics of the method.